LinkedList/Exercise/DLList.h

```diff
@@ -0,0 +1,355 @@
+#pragma once
+
+template<typename Type>
+class TDLList
+{
+private:
+
+	struct FNode
+	{
+		Type Data;
+		FNode* Next = nullptr;
+		FNode* Previous = nullptr;
+
+		FNode() = default;
+		FNode(const Type& NewData, FNode* NewPrevious, FNode* NewNext) : Data(NewData), Next(NewNext), Previous(NewPrevious) {};
+	};
+
+	FNode* Head;
+	FNode* Tail;
+
+	int Size;
+
+	FNode* GetNode(const int Index)
+	{
+		FNode* Current;
+
+		if (Index < (Size / 2))
+		{
+			Current = Head;
+			for (int i = 0; i < Index; i++)
+			{
+				Current = Current->Next;
+			}
+		}
+		else
+		{
+			Current = Tail;
+			for (int i = Size - 1; i > Index; i--)
+			{
+				Current = Current->Previous;
+			}
+		}
+		return Current;
+	}
+
+
+public:
+
+	class FIterator
+	{
+	private:
+		FNode* Current;
+
+	public:
+
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Data;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Data;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	//-------------------------------------------------------------------------------------
+
+	FIterator begin()
+	{
+		return FIterator(Head);
+	}
+
```

Comment:
No need fo spaces before and after the separators, however they look kinda cool.... so idk haha